File: QadeerApp.Web/Modules/ServerTypes/Administration/TrainingGradeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TrainingGradeRow } from "./TrainingGradeRow";

export interface TrainingGradeColumns {
    TrainingGradeId: Column<TrainingGradeRow>;
    Grade: Column<TrainingGradeRow>;
    TrainerName: Column<TrainingGradeRow>;
    TrainerNumber: Column<TrainingGradeRow>;
    ReferenceNumber: Column<TrainingGradeRow>;
    ScheduleType: Column<TrainingGradeRow>;
    CourseName: Column<TrainingGradeRow>;
    CourseCode: Column<TrainingGradeRow>;
    Department: Column<TrainingGradeRow>;
    Specialization: Column<TrainingGradeRow>;
    TrainingLevel: Column<TrainingGradeRow>;
    TrainingTerm: Column<TrainingGradeRow>;
    IsActive: Column<TrainingGradeRow>;
}

export class TrainingGradeColumns extends ColumnsBase<TrainingGradeRow> {
    static readonly columnsKey = 'Administration.TrainingGrade';
    static readonly Fields = fieldsProxy<TrainingGradeColumns>();
}

File: QadeerApp.Web/Modules/Administration/TrainingGrade/TrainingGradeColumns.cs
namespace QadeerApp.Administration.Columns;

[ColumnsScript("Administration.TrainingGrade")]
[BasedOnRow(typeof(TrainingGradeRow), CheckNames = true)]
public class TrainingGradeColumns
{
    [EditLink, AlignRight, Width(60)]
    public int TrainingGradeId { get; set; }

    [Width(100)]
    public string Grade { get; set; }

    [Width(150)]
    public string TrainerName { get; set; }

    [Width(120)]
    public string TrainerNumber { get; set; }

    [Width(120)]
    public string ReferenceNumber { get; set; }

    [Width(120)]
    public string ScheduleType { get; set; }

    [EditLink, Width(200)]
    public string CourseName { get; set; }

    [Width(120)]
    public string CourseCode { get; set; }

    [Width(160)]
    public string Department { get; set; }

    [Width(160)]
    public string Specialization { get; set; }

    [Width(120)]
    public string TrainingLevel { get; set; }

    [Width(160)]
    public string TrainingTerm { get; set; }

    [Width(80), QuickFilter]
    public short IsActive { get; set; }
}
